### PR TITLE
Fixed issue with combined k-eff estimator variance producing NaNs

### PR DESCRIPTION
**Pull request by nelsonag**
This PR fixes issue #613.  In that issue, @paulromano points out that sometimes (though probably very rarely) the combined k-eff variance estimator produces NaNs.  I also mentioned that the floating point exceptions also happen (every time) when using MG mode with survival biasing enabled.  

This bug definitely fixes the latter (MG mode + surv biasing), and was also written in a generic enough way to probably also fix the former problem.

The method was derived following the exact same process detailed in our current methodology (described in "Estimation and Interpretation of keff Confidence Intervals in MCNP" by Urbatsch et al), except two estimators were used instead of three in the reference.  I did the derivations by hand, but as a sanity check I also used a sympy Jupyter notebook to repeat the calculations.  The two methods produced perfectly equivalent formulas (phew).

That jupyter notebook is [here](https://gist.github.com/nelsonag/a76824eae14b749f455111697f3f6dc0).
I can scan in the hand derivations too if you *really* want them, but you probably wont be able to read them.

This PR also adds a test of the MG-mode survival biasing

### CODE REVIEW

**Comment by nelsonag** [comment]
Oh yeah, this PR will drop down to 2 rather small commits once #787 is merged (this is based off of that)

**Comment by paulromano** [comment]
@nelsonag The diff doesn't seem to be showing up correctly. Can you rebase this branch onto develop and force push?

**Comment by nelsonag** [comment]
@paulromano shold be good now.

File: src/eigenvalue.F90 [hunk 1]
```diff
@@ -458,73 +458,131 @@ subroutine calculate_combined_keff()
...
+
+    else
+      ! Use only two estimators
+      ! These equations are derived analogously to that done in the paper by
+      ! Urbatsch, but are simpler than for the three estimators case since the
+      ! block matrices of the three estimator equations reduces to scalars here
+
+      ! Store commonly used terms
+      f = cov(i, i) - cov(i, j)
+      g = (cov(i, i) + cov(j, j) - TWO * cov(i, j))
```
Comment (paulromano) [inline src/eigenvalue.F90]: extraneous oute rparentheses here

File: src/eigenvalue.F90 [hunk 2]
```diff
@@ -458,73 +458,131 @@ subroutine calculate_combined_keff()
...
+
+      ! Calculate combined estimate of k-effective
+      k_combined(1) = kv(i) - f / g * (kv(i) - kv(j))
+
+      ! Calculate standard deviation of combined estimate
+      k_combined(2) = sqrt(cov(i, i) * (ONE - f * f / g) * &
+           (ONE / n + (kv(i) - kv(j)) * (kv(i) - kv(j)) / ((n - ONE) * g)))
```
Comment (paulromano) [inline src/eigenvalue.F90]: I'm not able to get this to match your expression from the Jupyter notebook. More importantly, it doesn't contain N - 2 as I would expect. Check out Eq. 50 in [Urbatsch's technical report](https://laws.lanl.gov/vhosts/mcnp.lanl.gov/pdf_files/la-12658.pdf).

File: tests/test_mg_survival_biasing/test_mg_survival_biasing.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    def _build_inputs(self):
+        """Write input XML files."""
+        self._input_set.build_default_materials_and_geometry()
+        self._input_set.build_default_settings()
+        self._input_set.settings.survival_biasing = True
+        self._input_set.export()
+
+
+if __name__ == '__main__':
+    harness = MGBasicTestHarness('statepoint.10.*', False, mg=True)
```
Comment (paulromano) [inline tests/test_mg_survival_biasing/test_mg_survival_biasing.py]: statepoint.10.h5

File: src/eigenvalue.F90 (hunk 2 above)
Comment (nelsonag) [inline src/eigenvalue.F90]: You're right. Thanks for the different Urbatsch paper, I had the shorter article which went right to the 3-combined estimator.  I should be using something like Eq. 40 from the tech report you linked to (I think you meant 40 vice 50).  I will substitute that as soon as I understand the source of any differences.  Right now the differences are the (N-1) in the denominator of the right-most term, and the use of (N-2) in the front.  Once I understand that this should be easy.

Comment (paulromano) [inline src/eigenvalue.F90]: Yes, Eq. 40 is what I meant! Could have sworn I typed 40.

**Comment by paulromano** [comment]
I've confirmed that your fix does indeed avoid a floating point exception on the MG survival biasing problem. Nice going!

**Comment by paulromano** [comment]
Thanks @nelsonag for fixing this long-standing issue!

**Pull request merged**